### PR TITLE
Mid 114 add map to contagent extended card

### PR DESCRIPTION
**Pull request by BigTako**
Ticket: [this](https://trello.com/c/UP6JfACH/114-%D1%8E%D0%B7%D0%B5%D1%80-%D0%B2%D1%96%D0%B4%D0%BE%D0%B1%D1%80%D0%B0%D0%B6%D0%B5%D0%BD%D0%BD%D1%8F-%D0%BA%D0%B0%D1%80%D1%82%D0%B8-%D0%BD%D0%B0-%D0%B4%D0%B5%D1%82%D0%B0%D0%BB%D1%8C%D0%BD%D1%96%D0%B9-%D0%BA%D0%B0%D1%80%D1%82%D1%86%D1%96-%D1%81%D0%BF%D0%B5%D1%86%D1%96%D0%B0%D0%BB%D1%96%D1%81%D1%82%D0%B0-%D1%96-%D0%BE%D1%80%D0%B3%D0%B0%D0%BD%D1%96%D0%B7%D0%B0%D1%86%D1%96%D1%97)
image

### CODE REVIEW

File: src/app/_components/CardSpecialist/Specialist/ExtendedCard.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+import { addressesToPoints } from '@/utils/common';
 
 export function CardSpecialistExtended({ specialist, className }) {
+  const points = addressesToPoints(specialist.addresses);
   return (
     <CardModalWrapper className="mt-10 w-full max-w-[1080px] lg:mt-0 lg:w-[1080px]" key={specialist.id}>
-      <CardSpecialist specialist={specialist} className={className} extended />
+      <div className="flex flex-col gap-3 lg:gap-6">
+        <CardSpecialist specialist={specialist} className={className} extended />
+        {points && points.length > 0 && (
```
Comment (alex-popov-tech) [inline src/app/_components/CardSpecialist/Specialist/ExtendedCard.js]: points?.length

**Review by alex-popov-tech — APPROVED** [review]

**Review by Yivgha — APPROVED** [review]

File: src/app/(admin)/admin/_utils/common.js [hunk 2]
```diff
@@ -48,9 +48,12 @@ function transformAddresses({ addresses, type = 'create' }) {
 export const transformSupportFocuses = ({ focuses, focusesIds }) => {
   const focusesToUpdate = [];
   const focusesToCreate = [];
-  const focusesToDelete = toConnectList(focusesIds.filter(cutId => !focuses.some(focus => focus.id === cutId)));
 
-  focuses.forEach(focus => {
+  const focusesToDelete = toConnectList(
+    focusesIds.filter(cutId => !focuses?.some(focus => focus.id === cutId)) ?? true,
```
Comment (cherkasovnew) [inline src/app/(admin)/admin/_utils/common.js]: `focusesIds.filter(...)) ?? true` Do you think this makes any sense?
As I know the ["nullish coalescing operator"](https://uk.javascript.info/nullish-coalescing-operator) works with `null/undefined`, but [.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#return_value) always returns an array (empty or not)

File: src/app/_components/CardSpecialist/Organization/ExtendedCard.js [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
+import { addressesToPoints } from '@/utils/common';
 
 export function CardOrganizationExtended({ organization, className }) {
+  const points = addressesToPoints(organization.addresses);
   return (
     <CardModalWrapper className="mt-10 w-full max-w-[1080px] lg:mt-0 lg:w-[1080px]" key={organization?.id}>
-      <CardOrganization organization={organization} className={className} extended />
+      <div className="flex flex-col gap-3 lg:gap-6">
+        <CardOrganization organization={organization} className={className} extended />
+        {points && points.length > 0 && (
```
Comment (cherkasovnew) [inline src/app/_components/CardSpecialist/Organization/ExtendedCard.js]: `points && points.length > 0`  -> `points?.length > 0`?

File: src/app/_components/CardSpecialist/Specialist/ExtendedCard.js (hunk 1 above)
Comment (BigTako) [reply src/app/_components/CardSpecialist/Specialist/ExtendedCard.js]: Thanks for comment! Yep, seems better, fixed this.

File: src/app/_components/CardSpecialist/Organization/ExtendedCard.js (hunk 3 above)
Comment (BigTako) [reply src/app/_components/CardSpecialist/Organization/ExtendedCard.js]: Thanks for comment! Seems better, replaced with just `points?.length.`

File: src/app/(admin)/admin/_utils/common.js (hunk 2 above)
Comment (BigTako) [reply src/app/(admin)/admin/_utils/common.js]: Thanks for comment! Sorry, just typo here, should be like `!focuses?.some(focus => focus.id === cutId) ?? true`, fixed.

**Review by cherkasovnew — APPROVED** [review]

**Pull request merged**